lms/djangoapps/certificates/views/support.py

```diff
@@ -184,37 +184,35 @@ def regenerate_certificate_for_user(request):
     if response is not None:
         return response
 
-    # Check that the course exists
-    course = modulestore().get_course(params["course_key"])
-    if course is None:
-        msg = _("The course {course_key} does not exist").format(course_key=params["course_key"])
+    user = params["user"]
+    course_key = params["course_key"]
+
+    try:
+        get_course_overview(course_key)
+    except CourseOverview.DoesNotExist:
+        msg = _("The course {course_key} does not exist").format(course_key=course_key)
         return HttpResponseBadRequest(msg)
 
     # Check that the user is enrolled in the course
-    if not CourseEnrollment.is_enrolled(params["user"], params["course_key"]):
-        msg = _("User {username} is not enrolled in the course {course_key}").format(
-            username=params["user"].username,
-            course_key=params["course_key"]
+    if not CourseEnrollment.is_enrolled(user, course_key):
+        msg = _("User {user_id} is not enrolled in the course {course_key}").format(
+            user_id=user.id,
+            course_key=course_key
         )
         return HttpResponseBadRequest(msg)
 
     # Attempt to regenerate certificates
     try:
-        regenerate_user_certificates(params["user"], params["course_key"], course=course)
```

Comment:
Passing in the course is optional at this point and will be retrieved in the queue.py's `add_cert` function if it is None when generating the certificate (in v1 course certs).